### PR TITLE
Onboarding-Login and Signup Frontend Functionalitites

### PR DESCRIPTION
**Pull request by MGLarsenSW**
### Description
This pull request is the almost-complete frontend functionality of the relevant PBI of the Onboarding team.

### Changes
Changed the layout of the "Login", "Signup" pages, and added a "Welcome" page.
The Login and Signup pages communicate correctly with the backend, and as a result their functionalities work as intended.
The user cannot access certain pages if they do not have an authentication token, and vice versa cannot access Login and Signup if they have an authentication token.
Added a "Sign Out" functionality, which clears the token from the storage, once again preventing and allowing access to certain pages.
The Login page notifies the content creator if their login credentials are incorrect.
New content creators must now enter their password twice, in order to sign up properly.
Updated "hide password" icon.

### Related Issues
The layouts of Login, Signup, and Welcome are not properly implemented: their layout on different screens may vary. Additionally, some screens may find certain buttons inaccessible without the use of the tab button, 

### Checklist

- [ ]  Code has been tested locally and passes all relevant tests.
- [ ]  Documentation has been updated to reflect the changes, if applicable.
- [ ]  Code follows the established coding style and guidelines of the project.
- [ ]  All new and existing tests related to the changes have passed.
- [ ]  Any necessary dependencies or new packages have been properly documented.
- [x]  Pull request title and description are clear and descriptive.
- [x]  Reviewers have been assigned to the pull request.
- [x]  Any potential security implications have been considered and addressed.
- [ ]  Performance impact of the changes has been evaluated, if relevant.
### Notes for Reviewers
The styling of Login, Pages, and Welcome are not finished, and incredibly messy: some styling elements overlap in certain places, making for a lot of unnecessary code. Will be fixed in next sprint :)

### CODE REVIEW

**Comment by JackSMunn** [comment]
Not using PULL_REQUEST_TEMPLATE found in resources repo. Converting to draft, reopen when done

**Comment by Adachigaming** [comment]
### Description
This pull request is the almost-complete frontend functionality of the relevant PBI of the Onboarding team.

### Changes
Changed the layout of the "Login", "Signup" pages, and added a "Welcome" page.
The Login and Signup pages communicate correctly with the backend, and as a result their functionalities work as intended.
The user cannot access certain pages if they do not have an authentication token, and vice versa cannot access Login and Signup if they have an authentication token.
Added a "Sign Out" functionality, which clears the token from the storage, once again preventing and allowing access to certain pages.
The Login page notifies the content creator if their login credentials are incorrect.
New content creators must now enter their password twice, in order to sign up properly.
Updated "hide password" icon.


### Related Issues
The layouts of Login, Signup, and Welcome are not properly implemented: their layout on different screens may vary. Additionally, some screens may find certain buttons inaccessible without the use of the tab button, 

### Checklist

- [ ]  Code has been tested locally and passes all relevant tests.
- [ ]  Documentation has been updated to reflect the changes, if applicable.
- [ ]  Code follows the established coding style and guidelines of the project.
- [ ]  All new and existing tests related to the changes have passed.
- [ ]  Any necessary dependencies or new packages have been properly documented.
- [x]  Pull request title and description are clear and descriptive.
- [ ]  Reviewers have been assigned to the pull request.
- [x]  Any potential security implications have been considered and addressed.
- [ ]  Performance impact of the changes has been evaluated, if relevant.
### Notes for Reviewers
The styling of Login, Pages, and Welcome are not finished, and incredibly messy: some styling elements overlap in certain places, making for a lot of unnecessary code. Will be fixed in next sprint :)

File: app/src/services/auth.guard.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  } else {
+  React.useEffect(() => {
+    navigate("/profile")
+  }, []);
+}
+
+}
+
+
```
Comment (magnemo) [inline app/src/services/auth.guard.ts:34]: Functionerne skal have kommentar der følger JSDoc

File: app/src/pages/Courses.tsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ import { CubeTransparentIcon } from '@heroicons/react/24/outline';
 const Courses = () => {
   // States and Hooks
   const navigate = useNavigate();
-  const token = useToken();
+  const token = localStorage.getItem("token")
```
[the remark this answers is not in the document]
Comment (MGLarsenSW) [reply app/src/pages/Courses.tsx:22]: useToken er en gammel funktion som skabte konflikter. localStorage.getItem("token") checker om der er en token i localstorage som vi kan bruge

File: app/src/components/archive/Carousel.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import React, { useState, useEffect } from 'react';
```
[the remark this answers is not in the document]
Comment (Adachigaming) [reply app/src/components/archive/Carousel.tsx:1]: Carousel er en component der bliver brugt på Login, Signup og Welcome siderne: den indeholder functionaloteten bag teksten, der bliver vist på de ovenstående sider.

File: app/src/pages/Login.tsx [hunk 4]
```diff
@@ -25,69 +32,156 @@ const Login = () => {
...
+            <div className="text-center">
+                <span className=" text-gray-400 text-center text-base font-normal font-Montserrat gap-6">Ainda não tem conta? </span> {" "}
+                <Link to="/signup" className="text-neutral-700 text-base font-normal font-Montserrat underline hover:text-blue-500 gap-6">Cadastre-se agora</Link>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+</div>
+</main> )
```
[the remark this answers is not in the document]
Comment (Adachigaming) [reply app/src/pages/Login.tsx:184]: Vi beklager på vejne af vores kortsynethed...

File: app/src/pages/NotFound.tsx [hunk 5]
```diff
@@ -1,13 +1,53 @@
...
+          >
+            Voltar à página inicial
+          </button>
+        </Link>
+      </div>
+      </div>
+    </main>
+  );
+};
+
```
[the remark this answers is not in the document]
Comment (Adachigaming) [reply app/src/pages/NotFound.tsx:50]: ... Denne fejl vil fremover ikke blive begået på gensyn ...

File: app/src/pages/Signup.tsx [hunk 6]
```diff
@@ -40,77 +45,163 @@ const Signup = () => {
...
+        <Link to="/login" className="text-neutral-700 text-base font-normal font-Montserrat underline hover:text-blue-500 gap-6">Entre agora</Link>
       </div>
-    </main>
-  )
+  </div>
+  </div>
+  </div>
+</main>
+)
 }
```
[the remark this answers is not in the document]
Comment (Adachigaming) [reply app/src/pages/Signup.tsx:206]: ... Vi beklager for potentielle ulejligheder dette har forsaget...

File: app/src/pages/Welcome.tsx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  </div>
+</body>
+</div>
+</div>
+ 
+</main>
+    )
+}
+
+export default Welcome
```
[the remark this answers is not in the document]
Comment (Adachigaming) [reply app/src/pages/Welcome.tsx:82]: ... Du har hermed vores ord på, at dette problem vil blive afhjulpet i næste sprint.

File: app/src/services/auth.guard.ts (hunk 1 above)
Comment (Adachigaming) [reply app/src/services/auth.guard.ts:34]: Duly noted: will implement in further sprints.

File: app/src/pages/Courses.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (magnemo) [reply app/src/pages/Courses.tsx:22]: Kan i ikke lave en isusse der beskriver dette og hvad fixet er, så hvis nogle af de andre holde støder ind i dette problem, er der en kendt løsning

[the remark this answers is not in the document]
Comment (Adachigaming) [reply app/src/pages/Courses.tsx:22]: Issue er blevet oprettet.

File: app/src/services/auth.guard.ts (hunk 1 above)
Comment (SlovenlyTF) [reply app/src/services/auth.guard.ts:34]: I've made a comment in your backend pull request that explains how the comments should look like. But basically use Doxygen Documentation, if you didn't know😅

**Review by SlovenlyTF — CHANGES_REQUESTED** [review]

**Comment by JackSMunn** [comment]
FYI this PR is still failing checks.

**Comment by JackSMunn** [comment]
This is clearly not an active PR anymore. Last commit was 9th october. Closing. Reopen if you disagree.

**Pull request closed without merging**